macros/hook_functions/create_index.sql

```diff
@@ -0,0 +1,81 @@
+{% macro create_index_if_col_exists(table_model, column, unique=none, method=none) %}
+  {#
+    Create index function - allow setting a high level config by failing
+    gracefully if the column to be indexed does not exist. E.g. we can
+    configure all tables to have an index on "id", but not fail the dbt run if
+    any one table does not have an id column.
+
+    Arguments:
+      table_model   relation    e.g. {{ this }}
+      column        string      e.g. "id"
+      unique        boolean
+      method        string      e.g. "brin" -> needs to be valid index type
+
+    Notes:
+      1. arg column is not quoted, if your column required quoting it needs
+         to be quoted by the calling function, e.g. "'Id'"
+      2. make sure to use index types on compatible data types,
+         e.g. "brin" index type on an integer column will not fail the run
+         but results in a missing index
+
+    Sample usage:
+    {{ gemma_dbt_utils.create_index_if_col_exists(this, "id", false, "brin") }}
+  #}
+  {% if target.type == 'postgres' | as_bool() %}
+    {%- set hashing_query -%}
+    {# Hash index column to avoid running into postgres 63 byte identifier limit #}
+      SELECT md5('{{ table_model}}_{{ column }}')
+    {%- endset -%}
+
+    {%- if execute -%}
+      {%- set index_name = run_query(hashing_query).columns[0].values()[0] -%}
+    {%- else -%}
+    {# required to avoid compilation failure #}
```

Comment:
SQL Style (suggestion): Indent level :)

```suggestion
      {# required to avoid compilation failure #}
```